### PR TITLE
Adding flutter Gen package ✅

### LINKED ISSUE #184: Setup Flutter Gen to manage assets
Currently the code uses directly the asset path strings which are prone to a lot of issues (causing issues if that asset is deleted etc).
Screenshot 2024-12-06 at 2 30 04 PM

Use `flutter_gen` package to manage the assets paths
https://pub.dev/packages/flutter_gen

### PR DESCRIPTION
**Pull request by kareemabdeen**
## Description
Organized assets for seamless access through generated code.

### Motivation:
The integration of `flutter_gen` improves developer productivity and ensures reliable asset management.

Fixes #184 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (does not change functionality, e.g. code style improvements, linting)
## How Has This Been Tested?

- Ran `flutter pub run build_runner build` to generate the asset references.
- Verified that the auto-generated code correctly includes all assets.
- Tested asset usage in the app by replacing hardcoded asset paths with the `flutter_gen`-generated references.
- Ensured the app builds and runs without errors on both debug and release modes.

### Steps to Test:
1. Run `flutter pub run build_runner build` to generate the asset references.
2. Replace any asset path usages in the app with `Assets.<asset_path>`.
3. Run the app locally and ensure all assets (images, fonts, etc.) load correctly.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Tag the PR with the appropriate labels

---

Let me know if you need any further refinements!

### CODE REVIEW

**Comment by kareemabdeen** [comment]
Can you please review this pr @mdmohsin7

**Comment by kareemabdeen** [comment]
Hi @mdmohsin7,
I have rebased the current branch (feature/manage-assets-using-flutter-gen) with the updated develop branch, which now includes the latest changes from the recently merged PR. The branch is ready for testing and verification. Once all changes have been reviewed and validated, it can be merged.

File: lib/presentation/authentication/desktop/reset_password_view_desktop.dart [hunk 1]
```diff
@@ -44,10 +44,7 @@ class _ResetPasswordViewDesktopState extends State<ResetPasswordViewDesktop> {
             const SizedBox(
               height: 24,
             ),
-            SvgPicture.asset(
-              'assets/desktop/logo_desktop.svg',
-              width: 220,
```
Comment (mdmohsin7) [inline lib/presentation/authentication/desktop/reset_password_view_desktop.dart:49]: Hi @kareemabdeen, in few places the width and height has been removed for a few assets, can you pls check them?

File: lib/presentation/authentication/mobile/login_view_mobile.dart [hunk 2]
```diff
@@ -48,7 +49,7 @@ class _LoginViewMobileState extends State<LoginViewMobile> {
                 mainAxisAlignment: MainAxisAlignment.center,
                 crossAxisAlignment: CrossAxisAlignment.center,
                 children: [
-                  Image.asset(width: 136, height: 105, 'assets/logo_auth.png'),
```
Comment (mdmohsin7) [inline lib/presentation/authentication/mobile/login_view_mobile.dart:51]: Here the width has been omitted but the height is there

**Review by mdmohsin7 — CHANGES_REQUESTED** [review]
Please check the assets heights and widths

**Comment by kareemabdeen** [comment]
All is dn ✅ @mdmohsin7

**Review by mdmohsin7 — APPROVED** [review]
lgtm

**Pull request merged**